OpenResty LuaJIT / luajit 2.1 compatibility

Hi,

I just tried to build this with OpenResty's fork v2.1-20210510 and the ffi test in src/gen-compat.lua seems to return different results compared to LuaJIT 2.0.5.

https://github.com/DNS-OARC/dnsjit/blob/develop/src/gen-compat.lua#L23

Using OpenResty code path is used (ssize_t) while it isn't for LuaJIT (long).

At least on FreeBSD 13 that causes build issues while it builds if you many force the same code path as for LuaJIT.

```
--- core/compat.o ---
In file included from core/compat.c:25:
In file included from ./core/compat.hh:37:
In file included from /usr/include/unistd.h:39:
In file included from /usr/include/sys/types.h:49:
/usr/include/sys/_pthreadtypes.h:67:27: error: typedef redefinition with different types ('struct pthread *' vs 'struct pthread')
typedef struct  pthread                 *pthread_t;
                                         ^
./core/compat.hh:28:43: note: previous definition is here
typedef struct pthread { uint64_t a[1]; } pthread_t;
                                          ^
In file included from core/compat.c:25:
In file included from ./core/compat.hh:37:
In file included from /usr/include/unistd.h:39:
In file included from /usr/include/sys/types.h:49:
/usr/include/sys/_pthreadtypes.h:71:32: error: typedef redefinition with different types ('struct pthread_mutex *' vs 'struct pthread_mutex')
typedef struct  pthread_mutex           *pthread_mutex_t;
                                         ^
./core/compat.hh:24:49: note: previous definition is here
typedef struct pthread_mutex { uint64_t a[1]; } pthread_mutex_t;
                                                ^
In file included from core/compat.c:25:
In file included from ./core/compat.hh:37:
In file included from /usr/include/unistd.h:39:
In file included from /usr/include/sys/types.h:49:
/usr/include/sys/_pthreadtypes.h:73:31: error: typedef redefinition with different types ('struct pthread_cond *' vs 'struct pthread_cond')
typedef struct  pthread_cond            *pthread_cond_t;
                                         ^
./core/compat.hh:20:48: note: previous definition is here
typedef struct pthread_cond { uint64_t a[1]; } pthread_cond_t;
                                               ^
3 errors generated.
*** [core/compat.o] Error code 1

make[5]: stopped in /usr/ports/dns/dnsjit/work/dnsjit-1.2.1/src
1 error
```

This is on FreeBSD 13-STABLE and using dnsjit 1.2.1

--- a/src/core/compat.c
+++ b/src/core/compat.c
@@ -20,6 +20,4 @@
 
 #include "config.h"
 
-#include <stdint.h>
-
-#include "core/compat.hh"
+#include "core/compat.h"

--- a/src/gen-compat.lua
+++ b/src/gen-compat.lua
@@ -27,7 +27,6 @@ code, err = pcall(function()
     ]]
 end)
 if code then
-    print("#include <unistd.h>")
     print("typedef ssize_t luajit_ssize_t;")
 else
     print("typedef long luajit_ssize_t;")

